backend/src/email/routes/tests/email-transactional.routes.test.ts

```diff
@@ -55,179 +89,626 @@ describe('POST /transactional/email/send', () => {
+      subject: '\n\n\n',
+      body: '<script></script>',
+    }
+    const res = await request(app)
+      .post(endpoint)
+      .set('Authorization', `Bearer ${apiKey}`)
+      .send({
+        ...validApiCall,
+        subject: invalidHtmlTagsSubjectAndBody.subject,
+        body: invalidHtmlTagsSubjectAndBody.body,
+      })
+
+    expect(res.status).toBe(400)
+    expect(mockSendEmail).not.toBeCalled()
+
+    const transactionalEmail = await EmailMessageTransactional.findOne({
+      where: { userId: userId.toString() },
+    })
+    expect(transactionalEmail).not.toBeNull()
+    expect(transactionalEmail).toMatchObject({
+      recipient: validApiCall.recipient,
+      from: validApiCall.from,
+      status: TransactionalEmailMessageStatus.Unsent,
+    })
+    expect(transactionalEmail?.params).toMatchObject({
+      // NB sanitisation only occurs at sending step, doesn't affect saving in params
+      subject: invalidHtmlTagsSubjectAndBody.subject,
+      body: invalidHtmlTagsSubjectAndBody.body,
+      from: validApiCall.from,
+      replyTo: validApiCall.reply_to,
+    })
+    expect(transactionalEmail?.errorCode).toBe(EMPTY_MESSAGE_ERROR_CODE)
+  })
+
+  test('Should send email if subject and body are not empty after removing invalid HTML tags and metadata is saved correctly in db', async () => {
+    mockSendEmail = jest
       .spyOn(EmailService, 'sendEmail')
       .mockResolvedValue('message_id')
+    const invalidHtmlTagsSubjectAndBody = {
+      subject: '\nHELLO\n',
+      body: '<script> alert("hello") </script>',
+    }
 
     const res = await request(app)
-      .post('/transactional/email/send')
+      .post(endpoint)
       .set('Authorization', `Bearer ${apiKey}`)
       .send({
-        recipient: 'recipient@agency.gov.sg',
-        subject: 'subject',
-        body: '<p>body</p>',
-        from: 'Hello <invalid@agency.gov.sg>',
-        reply_to: user.email,
+        ...validApiCall,
+        subject: invalidHtmlTagsSubjectAndBody.subject,
+        body: invalidHtmlTagsSubjectAndBody.body,
       })
 
+    expect(res.status).toBe(202)
+    expect(mockSendEmail).toBeCalled()
+
+    const transactionalEmail = await EmailMessageTransactional.findOne({
+      where: { userId: userId.toString() },
+    })
+    expect(transactionalEmail).not.toBeNull()
+    expect(transactionalEmail).toMatchObject({
+      recipient: validApiCall.recipient,
+      from: validApiCall.from,
+      status: TransactionalEmailMessageStatus.Accepted,
+    })
+    expect(transactionalEmail?.params).toMatchObject({
+      // NB sanitisation only occurs at sending step, doesn't affect saving in params
+      subject: invalidHtmlTagsSubjectAndBody.subject,
+      body: invalidHtmlTagsSubjectAndBody.body,
+      from: validApiCall.from,
+      replyTo: validApiCall.reply_to,
+    })
+    expect(transactionalEmail?.errorCode).toBe(null)
+
+    expect(mockSendEmail).toBeCalledWith(
+      {
+        subject: '<br />HELLO<br />',
```

Comment:
realize that we actually insert `<br />` into the subject since we're using the same function `replaceNewLinesAndSanitize` for both subject and body. 

it doesn't affect the email sent, but just wanted to check if this is okay?

or if we should use two separate functions for sanitizing subject vs body